### PR TITLE
Tree delete endpoint

### PR DESCRIPTION
**Pull request by jgadling**
### Summary:
- **What:** Support deleting phylo runs / trees
- **Ticket:** [sc171781](https://app.shortcut.com/genepi/story/171781)
- **Env:** `<rdev link>`

### Notes:
Really, our current TODO list has a task to make sure we can delete phylo trees. I suspect (?) that just deleting a tree doesn't make sense and we actually want to delete the workflow that generated a tree, along with the tree itself. I think this is open to interpretation though - so any feedback is welcome here! 

### Checklist:
- [x] I merged latest `<base branch>`
- [x] I manually verified the change
- [x] I added labels to my PR
- [ ] I tested in multiple browsers
- [x] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

**Review by phoenixAja — APPROVED** [review]
This looks good to me

File: src/backend/aspen/api/views/phylo_runs.py [hunk 1]
```diff
@@ -182,3 +185,52 @@ async def kick_off_phylo_run(
...
+
+@router.delete("/{item_id}")
+async def delete_run(
+    item_id: int,
+    request: Request,
+    db: AsyncSession = Depends(get_db),
+    settings: Settings = Depends(get_settings),
+    user: User = Depends(get_auth_user),
+) -> bool:
+    item = await get_editable_phylorun_by_id(db, item_id, user)
```
Comment (phoenixAja) [inline src/backend/aspen/api/views/phylo_runs.py]: maybe rename `item` to `phylo_run`?

File: src/backend/aspen/api/views/phylo_runs.py [hunk 2]
```diff
@@ -182,3 +185,52 @@ async def kick_off_phylo_run(
...
+    item_db_id = item.id
+
+    for output in item.outputs:
```
Comment (phoenixAja) [inline src/backend/aspen/api/views/phylo_runs.py:232]: rename `output` to `phylo_tree`?

File: src/backend/aspen/api/views/tests/test_delete_phylo_runs.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from aspen.database.models import Group, PhyloTree, Sample, WorkflowStatusType
+from aspen.test_infra.models.phylo_tree import phylorun_factory, phylotree_factory
+from aspen.test_infra.models.sample import sample_factory
+from aspen.test_infra.models.sequences import uploaded_pathogen_genome_factory
+from aspen.test_infra.models.usergroup import group_factory, user_factory
+from aspen.test_infra.models.workflow import aligned_gisaid_dump_factory
+
+# All test coroutines will be treated as marked.
```
Comment (phoenixAja) [inline src/backend/aspen/api/views/tests/test_delete_phylo_runs.py:14]: what does this mean?

Comment (jgadling) [reply src/backend/aspen/api/views/tests/test_delete_phylo_runs.py:14]: it's a pytest thing - it means that all functions in this module that begin with the name `test_` will have the `asyncio` flag applied to them, and pytest will use the `pytest-asyncio` pugin to run them.

**Pull request merged**